stellargraph/core/graph.py

```diff
@@ -436,17 +666,59 @@ def to_adjacency_matrix(self, nodes: Optional[Iterable] = None):
         Returns:
              The weighted adjacency matrix.
         """
-        return self._graph.to_adjacency_matrix(nodes)
+        if self._graph is not None:
+            return self._graph.to_adjacency_matrix(nodes)
+
+        if nodes is None:
+            index = self._nodes._id_index
+            selector = slice(None)
+        else:
+            nodes = list(nodes)
+            index = ExternalIdIndex(nodes)
+            selector = np.isin(self._edges.sources, nodes) & np.isin(
+                self._edges.targets, nodes
+            )
+
+        weights = self._edges.weights[selector]
+        # these indices are computed relative to the index above. If `nodes` is None, they'll be the
+        # overall ilocs (for the original graph), otherwise they'll be the indices of the `nodes`
+        # list.
+        src_idx = index.to_iloc(self._edges.sources[selector])
+        tgt_idx = index.to_iloc(self._edges.targets[selector])
+        n = len(index)
+
+        adj = sps.csr_matrix((self._edges.weights, (src_idx, tgt_idx)), shape=(n, n))
```

Comment:
Is this meant to be:
```suggestion
        adj = sps.csr_matrix((weights, (src_idx, tgt_idx)), shape=(n, n))
```
and similarly below

Reply:
Good catch 👍